backend/src/email/middlewares/email.middleware.ts

```diff
@@ -79,22 +81,160 @@ const previewFirstMessage = async (
 
     if (!message) return res.json({})
 
-    const { body, subject, replyTo } = message
+    const { body, subject, replyTo, from } = message
     return res.json({
       preview: {
         body,
         subject,
         reply_to: replyTo,
+        from,
       },
     })
   } catch (err) {
     return next(err)
   }
 }
 
+/**
+ * Checks that the from address is either the user's email or the default app email
+ */
+const isFromAddressAccepted = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<Response | void> => {
+  const { from } = req.body
+  const userEmail = req.session?.user?.email
```

Comment:
This could be `undefined` if a user had logged in previously and the session has not expired. `500 Internal Server Error` will be thrown for the From Addresses routes.